packages/elements-core/src/utils/ref-resolving/resolvedObject.ts

```diff
@@ -0,0 +1,73 @@
+import { isArray, isPlainObject } from 'lodash';
+
+import { defaultResolver, ReferenceResolver } from './ReferenceResolver';
+
+const resolvedObjectSymbol = Symbol('ResolvedObject');
+const originalObjectSymbol = Symbol('OriginalObject');
+
+interface CreateResolvedObjectOptions {
+  contextObject?: object;
+  resolver?: ReferenceResolver;
+}
+
+export const createResolvedObject = (currentObject: object, options: CreateResolvedObjectOptions = {}): object =>
+  recursivelyCreateResolvedObject(currentObject, currentObject, [], options);
```

Comment:
Sucks so bad that we can't type this in a nicer way. I mean, I did make *some* improvement, but I honestly think it's not worth it for the complexity it adds:
```tsx
export const createResolvedObject = <K extends keyof any>(
  currentObject: Record<K, unknown>,
  options: CreateResolvedObjectOptions = {},
): K extends { $ref: string } ? Record<K, unknown> : object 
```

Reply:
Yeah, you would loose a lot of readability here for arguably little improvement.

We can still rethink that in the future I guess.